.buildkite/sanitizers.yml

```diff
@@ -0,0 +1,34 @@
+# These steps should only run on `sandbox.jl` machines, not `docker`-isolated ones
+# since we need nestable sandboxing.  The rootfs images being used here are built from
+# the `.buildkite/rootfs_images/llvm-passes.jl` file.
+agents:
+  queue: "julia"
+  # Only run on `sandbox.jl` machines (not `docker`-isolated ones) since we need nestable sandboxing
+  sandbox.jl: "true"
+  os: "linux"
+
+steps:
+  - label: "asan"
+    key: asan
+    plugins:
+      - JuliaCI/julia#v1:
+          version: 1.6
+      - staticfloat/sandbox#v1:
+          rootfs_url: https://github.com/JuliaCI/rootfs-images/releases/download/v1/llvm-passes.tar.gz
+          rootfs_treehash: "f3ed53f159e8f13edfba8b20ebdb8ece73c1b8a8"
+          uid: 1000
+          gid: 1000
+          workspaces:
+            - "/cache/repos:/cache/repos"
+      # `contrib/check-asan.jl` needs a `julia` binary:
+      - JuliaCI/julia#v1:
+          version: 1.6
+    commands: |
+      echo "--- Build julia-debug with ASAN"
+      contrib/asan.sh ./tmp/test-asan -j$${JULIA_NUM_CORES} debug
+      echo "--- Test that ASAN is enabled"
+      contrib/check-asan.jl ./tmp/test-asan/asan/usr/bin/julia-debug
```

Comment:
Can we put these things in `.buildkite/asan/` or something like that?  I feel the `contrib` directory is kind of the wild west, and I'd like to make it clearer that these things are used in CI, and that `asan.sh`, `check-asan.jl` and (potentially) `Make.user.asan` all "go together'.

Reply:
I was thinking `asan.sh` and `check-asan.jl` as scripts that some devs may invoke on their local checkout (rather than purely CI scripts). It'd be handy to have "just execute" script when you are first trying out how to use ASAN. It's also very useful for comparing your local setup against what's actually CI'ed. It ultimately doesn't matter if it's in `.buildkite` directory for such usecase but I think `contrib/` may be the "least bad" option for such use case. It may also be slightly more discoverable as it's easy to suspect utility scripts may be in there by the name `contrib/`.

What about moving them to `contrib/asan/build.sh` and  `contrib/asan/check.jl`? We can also create `contrib/asan/Make.user.tools` and `contrib/asan/Make.user.asan`.

Reply:
Yep, totally happy with that.